### ISSUE TITLE
Consider data-driven instead of code-driven event handling

### ISSUE THREAD

**Issue by meleneth** [Cnecomicz/gold-and-gallows#3]
Consider event handling like in https://github.com/meleneth/cnegng/blob/main/tests/ACME/test_pygame_event_handler.py

this is at the top layer, so this layer will convert key events (and eventually joystick events) to a virtualized 'controller' interface that maps the keys that are available on a xbox controller and keyboard keys to work the same such that game inputs work the same.

This lets you get to 'my control interface for any given thing is just this object that is configured' which then gets super simple to swap out in say on_enter_state methods

This module is in the ACME namespace under cnegng because it doesn't have any functionality that is directly wired into the game part of the game engine.

Which makes it generically useable, because you don't have to modify the base class to be able to make your handler functions do whatever is needed.

The docstrings are supposed to enable sphinx to extract and create good usage documentation, but it's not working yet

I especially like these lines

```
    def __call__(self, event):
        if (key_event := self.key_event_handlers.get(event.key)) is not None:
            key_event(event)
```

**Comment by meleneth** [comment]
I should point out that this doesn't actually implement the layer, it's the tooling to implement the layer, which I'm sure is confusing AF

**Comment by Cnecomicz** [comment]
Closing — completed. (Though I’ll probably continue to refine, the core idea is implemented in commit 2be9755.) Thanks @meleneth :) Now that I understand the pattern, I’ll probably look to see if this idea can be used in other places too — perhaps the update() and draw() methods too.
